data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -245,18 +242,9 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
 
     List<CloseableIterable<Record>> deletes = Lists.transform(posDeletes, this::openPosDeletes);
 
-    // if there are fewer deletes than a reasonable number to keep in memory, use a set
```

Comment:
When this logic was added a few years ago, we added position deletes into a set. We have been using bitmaps for a while now. In fact, vectorized reads always build bitmaps and have no threshold on the number of deletes. This has proven to work really well. Position deletes represented as bitmaps should always fit in memory.

Position deletes compress really well both on disk and in memory. We have seen this 100K threshold causing degradation in jobs without any good reason.

Reply:
Sweet, this makes sense and appreciate the simplification.  I remember seeing this threshold in the code and was wondering how we got to the 100000 value.